### PR TITLE
Add asset metadata to equippable data

### PR DESCRIPTION
**Pull request by ThunderDeliverer**
# Description

When getting the equippable slots, using `getEquippableSlotsFromParent` from `RMRKEquipRednerUtils` the child and parent asset were not being returned. As adding them is useful to the frontend developers, they were added.

# Actions

Extend `EquippableData` with `childMetadata` and `parentMetadata`.

## Bonus

The `assetsWithSlots` variable was renamed to `equippableData`.

# Checklist

- [x] Verified code additions
- [x] Updated NatSpec comments (if applicable)
- [x] Regenerated docs
- [x] Ran prettier
- [x] Added tests to fully cover the changes

### CODE REVIEW

**Review by steven2308 — COMMENTED** [review]
Can we call them `childAssetMetadata` and `parentAssetMetadata` instead?
Without it makes me think of `tokenURI`.

File: contracts/RMRK/utils/RMRKEquipRenderUtils.sol [hunk 1]
```diff
@@ -489,13 +493,15 @@ contract RMRKEquipRenderUtils is
      *      priority,
      *      parentCatalogAddress,
      *      isEquipped,
-     *      partMetadata
+     *      partMetadata,
+     *      childMetadata,
+     *      parentMetadata
```
Comment (ThunderDeliverer) [inline contracts/RMRK/utils/RMRKEquipRenderUtils.sol:498]:
```suggestion
     *      childAssetMetadata,
     *      parentAssetMetadata
```

File: contracts/RMRK/utils/RMRKEquipRenderUtils.sol [hunk 2]
```diff
@@ -537,21 +543,23 @@ contract RMRKEquipRenderUtils is
      *       priority
      *       parentCatalogAddress
      *       isEquipped
-     *       partMetadata
+     *       partMetadata,
+     *      childMetadata,
+     *      parentMetadata
```
Comment (ThunderDeliverer) [inline contracts/RMRK/utils/RMRKEquipRenderUtils.sol:548]:
```suggestion
     *       childAssetMetadata,
     *       parentAssetMetadata
```

File: contracts/RMRK/utils/RMRKEquipRenderUtils.sol [hunk 3]
```diff
@@ -586,11 +593,15 @@ contract RMRKEquipRenderUtils is
...
+            equippableData[i].partMetadata = IRMRKCatalog(parentAssetCatalog)
                 .getPart(tempAssetsWithSlots[i].slotPartId)
                 .metadataURI;
-            assetsWithSlots[i].parentAssetId = parentAssetId;
-            assetsWithSlots[i].parentCatalogAddress = parentAssetCatalog;
+            equippableData[i].parentAssetId = parentAssetId;
+            equippableData[i].parentCatalogAddress = parentAssetCatalog;
+            equippableData[i].childMetadata = IRMRKMultiAsset(childAddress)
+                .getAssetMetadata(childId, tempAssetsWithSlots[i].childAssetId);
+            equippableData[i].parentMetadata = IRMRKMultiAsset(parentAddress)
```
Comment (ThunderDeliverer) [inline contracts/RMRK/utils/RMRKEquipRenderUtils.sol:603]:
```suggestion
            equippableData[i].childAssetMetadata = IRMRKMultiAsset(childAddress)
                .getAssetMetadata(childId, tempAssetsWithSlots[i].childAssetId);
            equippableData[i].parentAssetMetadata = IRMRKMultiAsset(parentAddress)
```

File: test/renderUtils.ts [hunk 4]
```diff
@@ -345,7 +345,7 @@ describe('Advanced Equip Render Utils', async function () {
     ).to.eql([
       bn(0), // child Index
       [
-        // [Slot Id, asset Id, Asset priority]
+        // [Slot Id, asset Id, Asset priority, catalog address, isEquipped, partMetadata, childMetadata, parentMetadata]
```
Comment (ThunderDeliverer) [inline test/renderUtils.ts:348]:
```suggestion
        // [Slot Id, asset Id, Asset priority, catalog address, isEquipped, partMetadata, childAssetMetadata, parentAssetMetadata]
```

File: test/renderUtils.ts [hunk 5]
```diff
@@ -380,7 +386,7 @@ describe('Advanced Equip Render Utils', async function () {
     ).to.eql([
       bn(1), // child Index
       [
-        // [Slot Id, asset Id, Asset priority]
+        // [Slot Id, asset Id, Asset priority, catalog address, isEquipped, partMetadata, childMetadata, parentMetadata]
```
Comment (ThunderDeliverer) [inline test/renderUtils.ts:389]:
```suggestion
        // [Slot Id, asset Id, Asset priority, catalog address, isEquipped, partMetadata, childAssetMetadata, parentAssetMetadata]
```

File: test/renderUtils.ts [hunk 6]
```diff
@@ -416,7 +428,7 @@ describe('Advanced Equip Render Utils', async function () {
     ).to.eql([
       bn(2), // child Index
       [
-        // [Slot Id, asset Id, Asset priority]
+        // [Slot Id, asset Id, Asset priority, catalog address, isEquipped, partMetadata, childMetadata, parentMetadata]
```
Comment (ThunderDeliverer) [inline test/renderUtils.ts:431]:
```suggestion
        // [Slot Id, asset Id, Asset priority, catalog address, isEquipped, partMetadata, childAssetMetadata, parentAssetMetadata]
```

File: test/renderUtils.ts [hunk 7]
```diff
@@ -453,7 +471,7 @@ describe('Advanced Equip Render Utils', async function () {
...
```
Comment (ThunderDeliverer) [inline test/renderUtils.ts:474]:
```suggestion
        // [Slot Id, asset Id, Asset priority, catalog address, isEquipped, partMetadata, childAssetMetadata, parentAssetMetadata]
```

File: docs/RMRK/utils/RMRKEquipRenderUtils.md [hunk 8]
```diff
@@ -158,12 +158,12 @@ function getChildIndex(address parentAddress, uint256 parentId, address childAdd
...
 
 ```solidity
-function getEquippableSlotsFromParent(address targetChild, uint256 childId, uint64 parentAssetId) external view returns (uint256 childIndex, struct RMRKEquipRenderUtils.EquippableData[] assetsWithSlots)
+function getEquippableSlotsFromParent(address targetChild, uint256 childId, uint64 parentAssetId) external view returns (uint256 childIndex, struct RMRKEquipRenderUtils.EquippableData[] equippableData)
 ```
 
 Used to get the child&#39;s assets and slot parts pairs, identifying parts the said assets can be equipped into, for a specific parent asset.
 
-*Reverts if child token is not owned by an NFT.The full `EquippableData` struct looks like this:  [      slotPartId,      childAssetId,      parentAssetId,      priority,      parentCatalogAddress,      isEquipped,      partMetadata  ]*
+*Reverts if child token is not owned by an NFT.The full `EquippableData` struct looks like this:  [      slotPartId,      childAssetId,      parentAssetId,      priority,      parentCatalogAddress,      isEquipped,      partMetadata,      childMetadata,      parentMetadata  ]*
```
Comment (ThunderDeliverer) [inline docs/RMRK/utils/RMRKEquipRenderUtils.md:166]:
```suggestion
*Reverts if child token is not owned by an NFT.The full `EquippableData` struct looks like this:  [      slotPartId,      childAssetId,      parentAssetId,      priority,      parentCatalogAddress,      isEquipped,      partMetadata,      childAssetMetadata,      parentAssetMetadata  ]*
```

File: contracts/RMRK/utils/RMRKEquipRenderUtils.sol [hunk 9]
```diff
@@ -112,6 +114,8 @@ contract RMRKEquipRenderUtils is
         address parentCatalogAddress;
         bool isEquipped;
         string partMetadata;
+        string childMetadata;
+        string parentMetadata;
```
Comment (ThunderDeliverer) [inline contracts/RMRK/utils/RMRKEquipRenderUtils.sol:118]:
```suggestion
        string childAssetMetadata;
        string parentAssetMetadata;
```

File: contracts/RMRK/utils/RMRKEquipRenderUtils.sol [hunk 10]
```diff
@@ -103,6 +103,8 @@ contract RMRKEquipRenderUtils is
      * @return parentCatalogAddress Catalog address of the parent asset
      * @return isEquipped Whether the asset is currently equipped into the slot part or not
      * @return partMetadata Metadata URI of the given slot part
+     * @return childMetadata Metadata URI of the child asset
+     * @return parentMetadata Metadata URI of the parent asset
```
Comment (ThunderDeliverer) [inline contracts/RMRK/utils/RMRKEquipRenderUtils.sol:107]:
```suggestion
     * @return childAssetMetadata Metadata URI of the child asset
     * @return parentAssetMetadata Metadata URI of the parent asset
```

**Comment by ThunderDeliverer** [comment]
> Can we call them `childAssetMetadata` and `parentAssetMetadata` instead? Without it makes me think of `tokenURI`.

Makes sense, yeah. I renamed them ✅

**Review by steven2308 — APPROVED** [review]
Nice!

**Pull request merged**